src/analysis/special_functions/trigonometric/bounds.lean

```diff
@@ -120,4 +122,41 @@ begin
   simpa only [tan_zero, sub_zero, sub_pos] using mono zero_in_U x_in_U h1
 end
 
+lemma le_tan {x : ℝ} (h1 : 0 ≤ x) (h2 : x < π / 2) : x ≤ tan x :=
+begin
+  rcases eq_or_lt_of_le h1 with rfl | h1',
+  { rw tan_zero },
+  { exact le_of_lt (lt_tan h1' h2) }
+end
+
+lemma cos_lt_one_div_sqrt_sq_add_one {x : ℝ} (hx1 : -(π/2) ≤ x) (hx2: x ≤ π/2) (hx3 : x ≠ 0) :
```

Comment:
While you're at it, could you prove it for the bigger range of x between -pi and pi? The bound is trivial when cos is nnonpositive...

Reply:
Hmm. I'm not 100% keen, because I need this for an application involving bounding some integrals over `[0, π/2]` (this is a step in the proof of Euler's infinite product for the complex sine function). I already have to keep passing the proof that `-π/2 ≤ 0` as an argument; I'm not keen to have to include proofs that `π/2 ≤ π` as well.

Reply:
We should definitely have the more general version in mathlib, not the one restricted to your use case. If you want to state the more specialized version as a separate lemma, why not (but I'm not sure it's worth it).

Reply:
(And the more specialized version could be over `[0, π/2]` to spare you all proof arguments in your application)

Reply:
OK, I've added a proof covering the largest sensible range, `-3 * π / 2 ≤ x ≤ 3 * π / 2`.